index.js

```diff
@@ -253,7 +268,7 @@ async function run() {
       taskDefinitionFile :
       path.join(process.env.GITHUB_WORKSPACE, taskDefinitionFile);
     const fileContents = fs.readFileSync(taskDefPath, 'utf8');
-    const taskDefContents = maintainAppMeshConfiguration(removeIgnoredAttributes(cleanNullKeys(yaml.parse(fileContents))));
+    const taskDefContents = maintainAppMeshConfiguration(removeIgnoredAttributes(maintainEnvironmentVariables(cleanNullKeys(yaml.parse(fileContents)))));
```

Comment:
looks like `maintainAppMeshConfiguration` is doing a very similar thing. what do you think about refactoring it and your new func into one "maintainValidObjects" (or similar) func which checks for + reconstructs both cases? That way if we find more of these scenarios in the future we can embellish that function vs. add another nested call on this line

Reply:
I thought of doing it in the first place but then did not do it in order to keep two different functions separate for the better understanding. But these two functions are doing the same functionality to it does make sense to combine them together.